lib/video.ts

```diff
@@ -0,0 +1,53 @@
+import { createLocaleMap, getSortedLabels } from "./locale";
+import { Movie, Options } from "./options";
+
+export async function setupVideo(selector: string, options: Options) {
+  // Find the element.
+  const video = document.querySelector(selector);
+  if (!(video instanceof HTMLVideoElement)) {
+    throw new Error(`[wd-player] Can't find element "${selector}"`);
+  }
+
+  // Find and get movie data.
+  let movie: Movie;
+  try {
+    const res = await fetch(`${options.host}/${options.movie}/movie.json`);
+    movie = await res.json();
+  } catch (err) {
+    throw new Error(`[wd-player] Can't find movie "${options.movie}"`);
+  }
+
+  // Set necessary properties.
+  video.playsInline = true;
+  video.controls = true;
+  video.poster = options.poster;
+  video.dataset.poster = options.poster;
+
+  // Create sources.
+  for (const source of movie.sources) {
+    const element = document.createElement("source");
+    element.type = source.type;
+    element.src = source.src;
+
+    video.appendChild(element);
+  }
+
+  // Create captions that work for both maps and arrays.
+  let localeLabels = movie.captions ?? {};
+  if (Array.isArray(movie.captions)) {
+    localeLabels = createLocaleMap(movie.captions);
+  }
+
+  // We're reversing the array because of video.js doing something weird.
```

Comment:
lol nice

Reply:
Video.js traverses the tracks starting at the bottom going up, resulting in the order being flipped 😅